File: src/test/conditional_breakpoint_calls.py
import re
from util import *

send_gdb('b print_dot if checker()')
expect_gdb('Breakpoint 1')
send_gdb('watch dot_counter if checker()')
expect_gdb('Hardware watchpoint 2')
send_gdb('c')
expect_rr('EXIT-SUCCESS')

ok()

File: src/test/conditional_breakpoint_calls.c
/* -*- Mode: C; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */

#include "util.h"

int counter;
int dot_counter;

int checker(void) {
  ++counter;
  return 0;
}

static void print_dot(void) {
  atomic_printf(".");
  ++dot_counter;
}

int main(void) {
  int i;

  for (i = 0; i < 10; ++i) {
    print_dot();
  }

  atomic_puts("\nEXIT-SUCCESS");
  return 0;
}
